### ISSUE TITLE
enum.IntFlag now unpickles incorrectly on Python 3.11.4

### ISSUE THREAD

**Issue by da-woods** [python/cpython#105837]
# Bug report

Follow up to https://github.com/python/cpython/issues/105332

Following the changes to fix that issue `enum.IntFlag` appears to pickle correctly on the latest Python 3.12 commit (dd36656ada05731c069ecd5b1878380294fb1f3e) but fail on Python 3.11.4 (where on 3.11.2 it worked)

Code to reproduce

```
import enum
import pickle

class PxdEnum(enum.IntFlag):
    RANK_0 = 11
    RANK_1 = 37
    RANK_2 = 389

pickled = pickle.dumps(PxdEnum.RANK_2)

print(pickle.loads(pickled))
```

error output:

```
Traceback (most recent call last):
  File "<path>/enumtest.py", line 11, in <module>
    print(pickle.loads(pickled))
          ^^^^^^^^^^^^^^^^^^^^^
TypeError: unsupported operand type(s) for |: 'PxdEnum' and 'NoneType'
```

# Your environment

64 bit Linux, Python 3.11.4

**Comment by da-woods** [comment]
Just as background for this, in Cython we're converting C enums in `enum.IntFlags` when they're passed back to Python. One of the tests we have is that our converted enums can be pickled and unpickled, and this test is now failing. (Although it can be easily demonstrated outside Cython, as in the reproducible example above)

**Comment by sobolevn** [comment]
This was solved in https://github.com/python/cpython/pull/105348

Can you please verify?

**Comment by da-woods** [comment]
Sorry - I think you're right. I'd thought that the fix was in 3.11.4 but looking at the commit history the fix was just after 3.11.4.

I'll double check later today, but I suspect this is a false alarm caused by me not realising what has included in the release

**Comment by da-woods** [comment]
Yes - sorry - it's fixed on the Python 3.11 branch. I'd got slightly confused about what was in the 3.11.4 release.

I'll close this issue. Sorry for the noise.
